### PR TITLE
[misc] Normalize feature and disease ids before storing.

### PR DESCRIPTION
**Pull request by veronikaslc**
Resolving via HPO and MIM vocabularies

### CODE REVIEW

File: core/common/src/main/java/org/phenotips/remote/common/internal/api/DefaultJSONToMatchingPatientConverter.java [hunk 1]
```diff
@@ -43,22 +48,42 @@
...
 {
-    private Logger logger;
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultJSONToMatchingPatientConverter.class);
 
     private final Pattern hpoTerm; // not static: may be different from api version to api version
 
     private final Pattern disorderTerm;
 
-    private final Vocabulary ontologyService;
+    private static final Vocabulary MIM_VOCABULARY;
```
Comment (sdumitriu) [inline core/common/src/main/java/org/phenotips/remote/common/internal/api/DefaultJSONToMatchingPatientConverter.java]: Does this pass checkstyle? AFAIR, all constants should come before other types of fields.

Comment (veronikaslc) [inline core/common/src/main/java/org/phenotips/remote/common/internal/api/DefaultJSONToMatchingPatientConverter.java]: It does. just double-checked.

Comment (veronikaslc) [inline core/common/src/main/java/org/phenotips/remote/common/internal/api/DefaultJSONToMatchingPatientConverter.java]: we might have checks turned off?

File: core/client/src/main/java/org/phenotips/remote/client/internal/DefaultRemoteMatchingService.java [hunk 2]
```diff
@@ -240,9 +240,7 @@ public OutgoingMatchRequest getLastRequestSent(String patientId, String remoteSe
             return resultsList;
         }
 
-        DefaultJSONToMatchingPatientConverter patientConverter =
-            new DefaultJSONToMatchingPatientConverter(ApiConfiguration.LATEST_API_VERSION_STRING, logger,
-                ontologyService);
+        DefaultJSONToMatchingPatientConverter patientConverter = new DefaultJSONToMatchingPatientConverter();
```
Comment (allasm) [inline core/client/src/main/java/org/phenotips/remote/client/internal/DefaultRemoteMatchingService.java]: Why this change?
